packages/doenetml-worker-rust/lib-doenetml-core/src/components/doenet/graph.rs

```diff
@@ -5,20 +5,51 @@ use crate::props::UpdaterObject;
 #[component(name = Graph)]
 mod component {
 
-    use crate::general_prop::BooleanProp;
+    use crate::general_prop::{BooleanProp, NumberProp};
 
     enum Props {
         /// Whether the `<graph>` should be hidden.
         #[prop(value_type = PropValueType::Boolean, profile = PropProfile::Hidden)]
         Hidden,
         #[prop(value_type = PropValueType::AnnotatedContentRefs, profile = PropProfile::RenderedChildren)]
         RenderedChildren,
+        #[prop(value_type = PropValueType::Number, for_render, is_public)]
+        XMin,
+        #[prop(value_type = PropValueType::Number, for_render, is_public)]
+        XMax,
+        #[prop(value_type = PropValueType::Number, for_render, is_public)]
+        YMin,
+        #[prop(value_type = PropValueType::Number, for_render, is_public)]
+        YMax,
     }
 
     enum Attributes {
         /// Whether the `<graph>` should be hidden.
         #[attribute(prop = BooleanProp, default = false)]
         Hide,
+        #[attribute(prop = NumberProp, default = -10.0)]
+        XMin,
+        #[attribute(prop = NumberProp, default = 10.0)]
+        XMax,
+        #[attribute(prop = NumberProp, default = -10.0)]
+        YMin,
+        #[attribute(prop = NumberProp, default = 10.0)]
+        YMax,
+    }
+
+    #[derive(Debug, Clone, serde::Deserialize, serde::Serialize)]
+    #[cfg_attr(feature = "web", derive(tsify_next::Tsify))]
+    #[cfg_attr(feature = "web", tsify(from_wasm_abi))]
+    #[serde(expecting = "`x_min`, `x_max`, `y_min`, and `y_max` must be numbers")]
+    pub struct GraphChangeAxisLimitsActionArgs {
+        pub x_min: prop_type::Number,
+        pub x_max: prop_type::Number,
+        pub y_min: prop_type::Number,
+        pub y_max: prop_type::Number,
+    }
+
+    enum Actions {
+        ChangeAxisLimits(ActionBody<GraphChangeAxisLimitsActionArgs>),
```

Comment:
Do you consistently call these `AxisLimits`? To me they are more `ViewportBounds`

Reply:
Or, it could be `bounding box` in jsxgraph speak. I'm just thinking of it more in terms of the mathematical object than the widget. But, given that this is not visible by authors, I don't feel strongly about the name.

Reply:
OK